hikari/impl/stateful_cache.py

```diff
@@ -1598,3 +1605,27 @@ def update_voice_state(
         cached_voice_state = self.get_voice_state(voice_state.guild_id, voice_state.user_id)
         self.set_voice_state(voice_state)
         return cached_voice_state, self.get_voice_state(voice_state.guild_id, voice_state.user_id)
+
+    def delete_message(self, message_id: snowflakes.Snowflake) -> None:
+        if message_id in self._message_entries:
+            self._message_entries.pop(message_id)
+
+    def delete_messages(self, message_ids: typing.AbstractSet[snowflakes.Snowflake]) -> None:
+        for message_id in message_ids:
+            self._message_entries.pop(message_id)
+
+    def get_message(self, message_id: snowflakes.Snowflake) -> typing.Optional[messages.PartialMessage]:
+        return self._message_entries.get(message_id)
+
+    def set_message(self, message: messages.PartialMessage) -> None:
+        if len(self._message_entries) >= self.max_messages:
```

Comment:
rather than specifically handling this logic here it might make more sense to use collections.LimitedCapacityCacheMap here for the time being (this approach would also allow us to slot in different caching strategies instead of binding the logic here to a single approach).